ext/standard/basic_functions.stub.php

```diff
@@ -395,3 +395,199 @@ function assert($assertion, $description = null): bool {}
+
+/** @return int|false */
+function strcspn(string $str, string $mask, int $start = UNKNOWN, int $len = UNKNOWN) {}
+
+#if HAVE_NL_LANGINFO
+/** @return string|false */
+function nl_langinfo(int $item) {}
+#endif
+
+function strcoll(string $str1, string $str2): int {}
+
+function trim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function rtrim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function ltrim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function wordwrap(string $str, int $width = 75, string $break = "\n", bool $cut = false): string {}
+
+function explode(string $separator, string $str, int $limit = PHP_INT_MAX): array {}
+
+/**
+ * @param string $glue Optional - defaults to empty string
+ * @param array $pieces
+ */
+function implode($glue, $pieces): string {}
+
+/**
+ * @param string $str Optional - defaults to previous string
+ * @param string $token
+ * @return string|false
+ */
+function strtok($str, $token) {}
+
+function strtoupper(string $str): string {}
+
+function strtolower(string $str): string {}
+
+function basename(string $path, string $suffix = ""): string {}
+
+function dirname(string $path, int $levels = 1): string {}
+
+/** @return array|string */
+function pathinfo(string $path, int $options = UNKNOWN) {}
+
+/** @return string|false */
+function stristr(string $haystack, string $needle, bool $part = false) {}
+
+/** @return string|false */
+function strstr(string $haystack, string $needle, bool $part = false) {}
+
+/** @return int|false */
+function strpos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return int|false */
+function stripos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return int|false */
+function strrpos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return int|false */
+function strripos(string $haystack, string $needle, int $offset = 0) {}
+
+/** @return string|false */
+function strrchr(string $haystack, string $needle) {}
+
+function chunk_split(string $str, int $chunklen = 76, string $ending = "\r\n"): string {}
+
+/** @return string|false */
+function substr(string $str, int $start, int $length = UNKNOWN) {}
+
+/**
+ * @param mixed $str
+ * @param mixed $replace
+ * @param mixed $start
+ * @param mixed $length
+ * @return string|array|false
+ */
+function substr_replace($str, $replace, $start, $length = UNKNOWN) {}
+
+function quotemeta(string $str): string {}
+
+function ord(string $character): int {}
+
+function chr(int $codepoint): string {}
+
+function ucfirst(string $str): string {}
+
+function lcfirst(string $str): string {}
+
+function ucwords(string $str, string $delimiters = " \t\r\n\f\v"): string {}
+
+/** @param string|array $from */
+function strtr(string $str, $from, string $to = UNKNOWN): string {}
+
+function strrev(string $str): string {}
+
+function similar_text(string $str1, string $str2, float &$percent = UNKNOWN): int {}
```

Comment:
```suggestion
function similar_text(string $str1, string $str2, &$percent = null): int {}
```

Using null is generally okay as a default for ref params. Any value is ignored anyway.